File: docs-examples/pubsub-java/src/test/java/io/micronaut/configuration/lettuce/docs/pubsub/PlainTextBookListener.java
package io.micronaut.configuration.lettuce.docs.pubsub;

import io.micronaut.configuration.lettuce.pubsub.annotation.MessageChannel;
import io.micronaut.configuration.lettuce.pubsub.annotation.RedisListener;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Consumes;
import io.micronaut.messaging.annotation.MessageBody;

@RedisListener
class PlainTextBookListener {

    // tag::listener[]
    @Consumes(MediaType.TEXT_PLAIN)
    @MessageChannel("books.plain-text")
    void receive(@MessageBody String title) {
        System.out.println("Received plain text title " + title);
    }
    // end::listener[]
}

File: docs-examples/pubsub-python/src/test/python/micronaut/configuration/lettuce/docs/pubsub/PlainTextBookListener.py
from typing import Annotated

from micronaut.configuration.lettuce.pubsub.annotation import MessageChannel, RedisListener
from micronaut.http import MediaType
from micronaut.http.annotation import Consumes
from micronaut.messaging.annotation import MessageBody


@RedisListener
class PlainTextBookListener:

    # tag::listener[]
    @Consumes(MediaType.TEXT_PLAIN)
    @MessageChannel("books.plain-text")
    def receive(self, title: Annotated[str, MessageBody]) -> None:
        print(f"Received plain text title {title}")
    # end::listener[]
